bosh-director/lib/bosh/director/jobs/update_release.rb

```diff
@@ -484,25 +485,43 @@ def create_package(package_meta, release_dir)
 
       # @return [boolean] true if a new blob was created; false otherwise
       def save_package_source_blob(package, package_meta, release_dir)
-        return false unless package.blobstore_id.nil?
-
-        name, version, existing_blob = package_meta['name'], package_meta['version'], package_meta['blobstore_id']
+        name, version, existing_blob, sha1 = package_meta['name'], package_meta['version'], package_meta['blobstore_id'], package_meta['sha1']
         desc = "package '#{name}/#{version}'"
+        package_tgz = File.join(release_dir, 'packages', "#{name}.tgz")
+
+        if @fix
+          if package.blobstore_id != nil
+            logger.info("Verifying package #{desc} with blobstore_id: #{package.blobstore_id}")
+            return false if BlobUtil.verify_blob(package.blobstore_id, package.sha1)
```

Comment:
Seems like we are downloading blob for sha verification anyway, so what is the point of checking this (instead of just uploading every package)?

@loewenstein & @mariash